### PR TITLE
Fix wrong end state of trial

### PR DESCRIPTION
**Pull request by hougangliu**
Fixes: #694

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/695)

### CODE REVIEW

**Comment by johnugeorge** [comment]
@hougangliu Can you explain this PR?

**Comment by hougangliu** [comment]
> @hougangliu Can you explain this PR?

when `instance.IsCompleted() && instance.Spec.RetainRun` is True, https://github.com/kubeflow/katib/blob/master/pkg/controller/v1alpha2/trial/trial_controller.go#L284 will always run to mark the trial "Running" state even the Trial has completed.

File: pkg/controller/v1alpha2/trial/trial_controller.go [hunk 1]
```diff
@@ -281,8 +283,6 @@ func (r *ReconcileTrial) reconcileJob(instance *trialsv1alpha2.Trial, desiredJob
 		}
 	}
 
-	msg := "Trial is running"
-	instance.MarkTrialStatusRunning(TrialRunningReason, msg)
 	return deployedJob, nil
```
[the remark this answers is not in the document]
Comment (hougangliu) [reply pkg/controller/v1alpha2/trial/trial_controller.go]: yes, reconcileJob is called by reconcileTrial, reconcileTrial will use deployedJob status to update trial status

[the remark this answers is not in the document]
Comment (gaocegege) [reply pkg/controller/v1alpha2/trial/trial_controller.go]: OK, LGTM

**Pull request merged**